### ISSUE TITLE
Support function pointers

### ISSUE THREAD

**Issue by adetaylor** [google/autocxx#227]

**Comment by wmmc88** [comment]
Are all function pointers still not supported in autocxx?

**Comment by dougcooper** [comment]
As far as i can tell this is a limitation of the cxx library which does support passing closures with the inclusion and use of their cxx.h. However, autocxx inhibits its use and marks it as unsupported due to template specialization.

**Comment by adetaylor** [comment]
Yeah, sorry. I'd certainly be pleased to accept pull requests which can spot C++ functions that accept a [cxx `Fn`](https://cxx.rs/binding/fn.html) and then pass that through all our various pipeline stages to feed the right sort of thing to `cxx` in our generated bindings. I suspect it might be tricky - I imagine the information we get from bindgen is incomplete or insufficient for this, but you never know. A good start would be to add a test to `integration_tests.rs` and then [use these instructions to get logging out](https://github.com/google/autocxx/blob/main/book/src/contributing.md#reporting-bugs), paying particular attention to what information `bindgen` gives to `autocxx` at the start of the process.
